src/Hcal/HcalRecProducer.cxx

```diff
@@ -0,0 +1,191 @@
+/**
+ * @file HcalRecProducer.cxx
+ * @brief Class that performs basic HCal reconstruction
+ * @author Tom Eichlersmith, University of Minnesota
+ * @author Cristina Suarez, Fermi National Accelerator Laboratory
+ */
+
+#include "Hcal/HcalRecProducer.h"
+
+namespace hcal {
+
+    HcalRecProducer::HcalRecProducer(const std::string& name, framework::Process& process) :
+        Producer(name, process) {
+    }
+
+    HcalRecProducer::~HcalRecProducer() { }
+
+    void HcalRecProducer::configure(framework::config::Parameters& ps) {
+
+        //collection names
+        digiCollName_    = ps.getParameter<std::string>("digiCollName");
+        digiPassName_    = ps.getParameter<std::string>("digiPassName");
+        simHitCollName_  = ps.getParameter<std::string>("simHitCollName");
+        simHitPassName_  = ps.getParameter<std::string>("simHitPassName");
+        recHitCollName_  = ps.getParameter<std::string>("recHitCollName");
+
+        mip_energy_      = ps.getParameter<double>( "mip_energy"  );
+	pe_per_mip_      = ps.getParameter<double>( "pe_per_mip"   );
+        clock_cycle_     = ps.getParameter<double>( "clock_cycle"    );
+        voltage_per_mip_ = ps.getParameter<double>( "voltage_per_mip" );
+	pedestal_        = ps.getParameter<double>( "pedestal" );
+	gain_            = ps.getParameter<double>( "gain" );
+	attlength_       = ps.getParameter<double>("attenuationLength");
+    }
+
+    void HcalRecProducer::produce(framework::Event& event) {
+        // Get the Hcal Geometry
+        const auto& hcalGeometry = getCondition<ldmx::HcalGeometry>(ldmx::HcalGeometry::CONDITIONS_OBJECT_NAME);
+
+        std::vector<ldmx::HcalHit> hcalRecHits;
+        auto hcalDigis = event.getObject<ldmx::HgcrocDigiCollection>( digiCollName_ , digiPassName_ );
+        int numDigiHits = hcalDigis.getNumDigis();
+
+        // Loop through digis
+	int iDigi = 0;
+	while( iDigi < numDigiHits ) {
+	  auto digi = hcalDigis.getDigi( iDigi );
+
+	  // ID from first digi sample
+	  ldmx::HcalDigiID digiId(digi.id());
+	  ldmx::HcalID id(digiId.section(),digiId.layer(),digiId.strip());
+	  
+	  // Position from ID (returns -99999 for unknown x or y coordinate)
+	  double x,y,z;
+	  hcalGeometry.getStripAbsolutePosition( id , x , y , z );
```

Comment:
See comment in parent PR about removing this relic inherited from EcalHexReadout.